Debugger.py

```diff
@@ -73,42 +74,83 @@ def parseCommandLineArgs():
-    :return: This does not return. Simply raises ValueError in cases where paths are not valid.
+    This function will simply validate file existence and file extension.
+    If either is not true, then will raise an error.
+    :param filename: str, file path to be validated
+    :param ext: str, expected extension
+    :return: boolean True,
     """
-    gotSymbols = False
-    if not os.path.exists(argsWithPaths.input):
-        raise ValueError("ERROR: file {} does not exists.".format(argsWithPaths.input,))
+    if not os.path.exists(filename):
+        raise ValueError("ERROR: File {} does not exists.".format(filename,))
     else:
-        if os.path.exists(argsWithPaths.input.split(".")[0] + ".sym"):
-            gotSymbols = True
+        fileExt = filename.split(".")[-1]
+        if fileExt != ext:
+            raise ValueError("ERROR: Incorrect file extension on file {}".format(filename,))
 
-    return gotSymbols
+    return True
+
+
+def validateBreakPointFile(usable_args, ext=None):
+    """
+    This function will check the inputted breakpoint file and handle the cases.
+    If either is not true will raise error.
+    :param usable_args: An input parsed object as provided by argparse.parse_args()
+    :param ext: str, expected extension for breakpoint file
+    :return inputBreakpointFile: str, name of the breakpointFile that will be used by Debugger
+    """
+    inputBreakpointFile = usable_args.breakpoint
+
+    if inputBreakpointFile is None:
+        inputBreakpointFile = usable_args.input.split(".")[0]+".bp"
+        Path(inputBreakpointFile).touch()
+        return inputBreakpointFile
+    elif not os.path.exists(inputBreakpointFile):
+        inputFileExt = inputBreakpointFile.split(".")[-1]
+        if inputFileExt != ext:
+            raise ValueError("ERROR: Incorrect file extension on Breakpoint file {}".format(inputBreakpointFile,))
+        Path(inputBreakpointFile).touch()
+        return inputBreakpointFile
+    elif os.path.exists(inputBreakpointFile):
+        inputFileExt = inputBreakpointFile.split(".")[-1]
+        if inputFileExt != ext:
+            raise ValueError("ERROR: Incorrect file extension on Breakpoint file {}".format(inputBreakpointFile,))
+        return inputBreakpointFile
 
 
 if __name__ == '__main__':
     usableArgs = parseCommandLineArgs()
 
+    goodInputFile = False
     gotSymbols = False
     symbolsFile = None
+    breakpointFile = None
 
     if usableArgs.input is not None:
-        gotSymbols = validatePaths(usableArgs)  # Make sure the parsed info is usable before using it!
-        symbolsFile = usableArgs.input.split(".")[0] + ".sym" if gotSymbols else ""
+        # Make sure the parsed info is usable before using it!
+        goodInputFile = validateFilePath(usableArgs.input, ext="bin")
+        gotSymbols = validateFilePath(usableArgs.input.split(".")[0] + ".sym", ext="sym")
+        breakpointFile = validateBreakPointFile(usableArgs, ext="bp")
```

Comment:
This will result in a break point file being used no matter if the user wants it or not.